addons/viewport/src/preview/withViewport.js

```diff
@@ -9,38 +9,41 @@ import {
 } from '../shared';
 
 function noop() {}
-let handler = noop;
 
-const subscription = () => {
+const subscription = handler => () => {
```

Comment:
I noticed an issue when you create 2 stories with `subscription`/`onViewportChange`, the 2nd one receives the event twice every time you change the viewport, which should not be

Reply:
Now it creates a new subscription on each rerender (`subscription(handler)`) is a new function each time

Reply:
You may want to store `currentHandler` as module-level variable instead

Reply:
Can you please add a second example with `onViewportChange` so that everyoune could reproduce the issue

Reply:
Oh I see, changing of `handler` variable prevents the subscription from removing. This should fix it without affecting performance:

```
const subscription = () => {
  const currentHandler = handler
  const channel = addons.getChannel();
  channel.on(VIEWPORT_CHANGED_EVENT_ID, currentHandler);
  return () => channel.removeListener(VIEWPORT_CHANGED_EVENT_ID, currentHandler);
};
```

Reply:
Yep, this should work.

Reply:
@Hypnosphi do you still need the second example?

Reply:
As you wish. Or maybe a testcase would be better, just to make sure it doesn't break in future

Reply:
I agree, test case is even better. Will add it